### PR TITLE
Elysium changes

### PR DESCRIPTION
**Pull request by NicholasLYang**
Mostly changes to connect media to users and projects, while getting uploading finished up.

### CODE REVIEW

File: src/routes/media.rs [hunk 1]
```diff
@@ -109,3 +138,17 @@ fn process_entries(
         )),
     }
 }
+
+fn get_file_ext(file_fields: &Vec<multipart::server::SavedField>) -> Result<&str, Custom<String>> {
```
Comment (A1Liu) [inline src/routes/media.rs:139]: Is it better to have the error contain a `Custom<String>` or a `Custom<&str>` (where the latter doesn't use the `.to_string()` method of string literals)?

File: src/routes/media.rs [hunk 2]
```diff
@@ -67,11 +67,41 @@ fn process_file_upload(
     }
 }
 
+fn get_foreign_key(
+    key_name: &'static str,
```
Comment (A1Liu) [inline src/routes/media.rs:71]: why is this a static reference?

File: src/routes/media.rs [hunk 3]
```diff
@@ -67,11 +67,41 @@ fn process_file_upload(
...
+    entries: &Entries,
+) -> Result<Option<i32>, Custom<String>> {
+    let id_field = entries.fields.get(key_name).map(|field| &field[0].data);
+
+    // This isn't a map because we need to return out if parse fails
+    Ok(match id_field {
```
Comment (A1Liu) [inline src/routes/media.rs]: I think this block should be a little less nested; it's kinda hard to reason about its behavior right now.

File: src/resolvers/project.rs [hunk 4]
```diff
@@ -68,4 +87,11 @@ impl Project {
             .load::<User>(conn)
             .expect("Could not load contributors")
     }
+
+    pub fn media(&self, conn: &diesel::PgConnection) -> Vec<Medium> {
+        media::table
+            .filter(media::project_id.eq(self.id))
+            .load::<Medium>(conn)
+            .expect("Could not load media")
```
Comment (A1Liu) [inline src/resolvers/project.rs:104]: Is a `.expect` ok here? I'm guessing this is code that won't be in the release version, but I wanna make sure I'm not missing anything. Also, maybe we should flag this kind of thing? Something like `// @fix_later`

**Review by A1Liu — CHANGES_REQUESTED** [review]
I left comments on code style and clarity, and tested to make sure it builds (it does).

File: src/resolvers/project.rs (hunk 4 above)
Comment (NicholasLYang) [reply src/resolvers/project.rs:104]: Eh, I suppose we could do a Result but I'm fine with this being a fatal error.

File: src/routes/media.rs (hunk 2 above)
Comment (NicholasLYang) [reply src/routes/media.rs:71]: Because it's a string literal. I could make it a regular slice but why bother?

File: src/routes/media.rs (hunk 3 above)
Comment (NicholasLYang) [reply src/routes/media.rs]: Alright, sure

File: src/routes/media.rs (hunk 1 above)
Comment (NicholasLYang) [reply src/routes/media.rs:139]: Yeah, just tried this out rn. With string refs you need to add a lifetime and tie it to some resource. Better to just move it.

Comment (A1Liu) [reply src/routes/media.rs:139]: Ok makes sense. Would it make sense to use a static lifetime there?

Comment (NicholasLYang) [reply src/routes/media.rs:139]: I considered that but then `format!` gets messed up.

**Review by A1Liu — APPROVED** [review]
Looks good to me!

**Pull request merged**